### ISSUE TITLE
`ElementId` type

### ISSUE THREAD

**Issue by nsaunders** [purescript-web/purescript-web-dom#57]
### Description

After reading [an old comment](https://github.com/purescript-halogen/purescript-halogen/issues/130#issuecomment-270888416) from a Halogen maintainer, it seems to me that the lack of a newtype wrapper for a value of the element `id` property/attribute could be considered _a missing feature that, had it existed, would have been ported to [web-html](https://github.com/purescript-web/purescript-web-html) along with [`ClassName`](https://pursuit.purescript.org/packages/purescript-web-html/4.1.0/docs/Web.HTML.Common#t:ClassName), [`AttrName`](https://pursuit.purescript.org/packages/purescript-web-html/4.1.0/docs/Web.HTML.Common#t:AttrName), and [`PropName`](https://pursuit.purescript.org/packages/purescript-web-html/4.1.0/docs/Web.HTML.Common#t:PropName)_.

At the same time, in #56 I suggest that perhaps those types belong in this library instead of web-html, which is why I am creating the issue here.

In short, I am suggesting that, if #56 is accepted, an `ElementId` type ought to be added as well.

### Prerequisites

- [x] Before opening an issue, please check the DOM standard (https://dom.spec.whatwg.org/).

> **Note**
> None of the newtype wrappers mentioned above, including the proposed `ElementId`, are mentioned in the spec, but neither are they in the [HTML standard](https://html.spec.whatwg.org/).

**Comment by garyb** [comment]
Seems reasonable to me :+1:. I'd probably lean towards `ElementId`, although I don't have any reasoned justification for it.

**Comment by nsaunders** [comment]
Thanks @garyb, I was on the fence myself, so I've updated the proposed type name to `ElementId`.
